backend/src/main/kotlin/app/ehrenamtskarte/backend/cards/webservice/schema/CardMutationService.kt

```diff
@@ -156,10 +159,16 @@ class CardMutationService {
             userEntitlements.regionId.value
         )
 
-        val activationCode = CardCreationResultModel(
-            createDynamicActivationCode(updatedCardInfo, userId = null),
-            if (generateStaticCode) createStaticVerificationCode(updatedCardInfo, userId = null) else null
-        )
+        val activationCode = transaction {
+            val revokedCount = CardRepository.revokeByEntitlementsId(userEntitlements.id.value)
+            if (revokedCount > 0) {
+                logger.info("Revoked {} cards associated with the user entitlements {}", revokedCount, userEntitlements.id.value)
```

Comment:
I assume we want to revoke old cards and not delete, or?

Reply:
we can think about some cleanup functionality at a later point. I'm not sure what happens at the moment if a user has a activated on the phone and the card entry gets deleted